Paco-Server/ear/default/web/js/viz/vizService.js

```diff
@@ -0,0 +1,148 @@
+pacoApp.factory('experimentsVizService', ['$http', 'experimentService', function ($http, experimentService) {
+
+  var experiment = '';
+  var events = ' ';
+
+  function getExperiment(id) {
+
+    var getExperiment = experimentService.getExperiment(id).then(function successCallback(experimentData) {
+      return experimentData.data;
+    }, function errorCallback(error) {
+      return error;
+    });
+    experiment = getExperiment.then(function successCallback(experiment) {
+      return experiment;
+    });
+    return experiment;
+  }
+  function getParticipants(experimentId){
+
+   if(experimentId != undefined){
+         var message = '{ "select":["who"], "query" : { "criteria" : "experiment_id = ?", "values" : ['+experimentId+']},"group":"who"}';
+   }
+    var participants = $http({
+      method: 'POST',
+      url: '/csSearch',
+      data: angular.fromJson(message),
+    }).then(function successCallback(response) {
+      return response;
+    }, function errorCallback(error) {
+      return error;
+    });
+    return participants;
+
+  }
+
+  function getEventCounts(id){
+    var getCount = $http({
+      method: 'GET',
+      url: '/participantStats?experimentId=' + id + '&reportType=totalEventCounts&statv2=1'
+    }).then(function successCallback(response) {
+      return response;
+    }, function errorCallback(error) {
+      return error;
+    });
+    return getCount;
+  }
+
+  function getEvents(experimentId, group, participants, startDate, startTime, endDate, endTime) {
+
+    var message = "";
```

Comment:
This method can be simplified significantly. Also, the if clauses are not mutually exclusive which means the "message" object  may get built and replaced multiple times.